lib/web.ts

```diff
@@ -436,6 +479,91 @@ export default class AutoConsent {
     return this.domActions.undoPrehide();
   }
 
+  /**
+   * Apply cosmetic filters
+   * @returns true if the filters were applied, false otherwise
+   */
+  async applyCosmeticFilters(styles?: string) {
+    performance.mark('autoconsent-apply-filterlist-start');
+    if (!this.filtersEngine) {
+      return false;
+    }
+    const logsConfig = this.config?.logs;
+    if (!styles) {
+      styles = getCosmeticStylesheet(this.filtersEngine);
+    }
+    this.updateState({ cosmeticFiltersOn: true });
+    try {
+      this.cosmeticStyleSheet = await this.domActions.createOrUpdateStyleSheet(styles, this.cosmeticStyleSheet);
+      logsConfig?.lifecycle && console.log("[cosmetics]", this.cosmeticStyleSheet, location.href);
+      document.adoptedStyleSheets.push(this.cosmeticStyleSheet);
+      performance.mark('autoconsent-apply-filterlist-end');
+    } catch (e) {
+      this.config.logs && console.error('Error applying cosmetic filters', e);
+      performance.mark('autoconsent-apply-filterlist-end');
+      return false;
+    }
+    return true;
+  }
+
+  undoCosmetics() {
+    this.updateState({ cosmeticFiltersOn: false });
+    this.domActions.removeStyleSheet(this.cosmeticStyleSheet);
+  }
+
+  filterListFallback() {
+    if (!this.filtersEngine) {
+      this.updateState({ lifecycle: 'nothingDetected' });
+      return false;
+    }
+
+    const cosmeticStyles = getCosmeticStylesheet(this.filtersEngine);
+
+    // this may be a false positive: sometimes filters hide unrelated elements that are not cookie pop-ups
```

Comment:
note that we still need to do this check to know if we should report to the background script